src/gsi/read_radarref_mosaic_directDA.f90

```diff
@@ -0,0 +1,490 @@
+!
+!   output argument list:
+!     nread    - number of type "obstype" observations read
+!     ndata    - number of type "obstype" observations retained for further processing
+!     nobs     - array of observations on each subdomain for each processor
+!
+! USAGE:
+!   INPUT FILES:  refInGSI
+!
+!   OUTPUT FILES:
+!
+! REMARKS:
+!
+! ATTRIBUTES:
+!   LANGUAGE: FORTRAN 90
+!   MACHINE:  Linux cluster(Wjet)
+!
+!$$$
+!
+!_____________________________________________________________________
+!
+  use kinds, only: r_kind,r_double,i_kind
+  use constants, only: zero,one,rad2deg,deg2rad
+  use convinfo, only: nconvtype,ctwind,cgross,cermax,cermin,cvar_b,cvar_pg, &
+        ncmiter,ncgroup,ncnumgrp,icuse,ictype,icsubtype,ioctype
+  use gsi_4dvar, only: l4dvar,l4densvar,winlen
+  use gridmod, only: nlon,nlat,nlon_regional,nlat_regional,txy2ll,tll2xy
+  use mod_wrfmass_to_a, only: wrfmass_obs_to_a8
+  use mpimod, only: npe
+  use obsmod, only: perturb_obs
+  use directDA_radaruse_mod, only: oe_dbz, refl_lowbnd_dbz
+
+  implicit none
+!
+
+  character(10),    intent(in)    :: infile,obstype
+  integer(i_kind),  intent(in)    :: lunout
+  integer(i_kind),  intent(inout) :: nread,ndata
+  integer(i_kind),  intent(inout) :: nodata
+  integer(i_kind),dimension(npe) ,intent(inout) :: nobs
+  real(r_kind),     intent(in   ) :: twind
+  character(20),    intent(in)    :: sis
+
+! Declare local parameters
+  real(r_kind),parameter:: r90  = 90.0_r_kind
+  real(r_kind),parameter:: r360 = 360.0_r_kind
+
+!
+!  For reflectiivty mosaic
+!
+  integer(i_kind) nreal,nchanl
+
+  integer(i_kind) ifn,i
+ 
+  real(r_kind)  :: maxref
+  integer(i_kind) :: ilon,ilat
+
+  logical :: nsslrefobs
+  logical dbzob
+! real(r_kind),allocatable,dimension(:,:):: cdata_all
+  real(r_kind),allocatable,dimension(:,:):: cdata_out
+  real(r_kind) :: dbzerr, thiserr
+! real(r_kind) :: rmins_an,rmins_ob                                                     
+  real(r_kind) :: hgt_ref(33)
+  data hgt_ref / 500.0,   750.0,  1000.0,  1250.0,  1500.0,  1750.0,  2000.0,   &
+                2250.0,  2500.0,  2750.0,  3000.0,  3500.0,  4000.0,  4500.0,   &
+                5000.0,  5500.0,  6000.0,  6500.0,  7000.0,  7500.0,  8000.0,   &
+                8500.0,  9000.0, 10000.0, 11000.0, 12000.0, 13000.0, 14000.0,   &
+               15000.0, 16000.0, 17000.0, 18000.0, 19000.0/
+
+  real(r_kind) :: i_maxloc,j_maxloc,k_maxloc
+  integer(i_kind) :: kint_maxloc
+  real(r_kind) :: dbz_max
+  integer(i_kind) :: ndata2
+
+!
+!  for read in bufr 
+!
+    real(r_kind) :: hdr(5),obs(1,35)
+    character(80):: hdrstr='SID XOB YOB DHR TYP'
+    character(80):: obsstr='HREF'
+
+    character(8) subset
+    character(8) station_id
+    real(r_double) :: rstation_id
+    equivalence(rstation_id,station_id)
+    integer(i_kind)  :: lunin,idate
+    integer(i_kind)  :: ireadmg,ireadsb
+
+    integer(i_kind)  ::  maxlvl
+    integer(i_kind)  ::  numlvl,numref,numobsa,nmsgmax,maxobs
```

Comment:
The variable, numobsa, has been identified as unused.  Please remove unused variable.

Reply:
We removed them as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-1833e300b7dd38188cb415090239f3f184b0ad14deaae73fb732e4fa73d4f221R125